homeassistant/components/life360/helpers.py

```diff
@@ -1,7 +1,208 @@
 """Life360 integration helpers."""
-from life360 import Life360
 
+from __future__ import annotations
 
-def get_api(authorization=None):
+from collections.abc import Callable
+from typing import Any, TypedDict
+
+from life360 import Life360, Life360Error, LoginError
+
+from homeassistant.const import (
+    ATTR_BATTERY_CHARGING,
+    ATTR_BATTERY_LEVEL,
+    ATTR_ENTITY_PICTURE,
+    ATTR_GPS_ACCURACY,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    ATTR_NAME,
+    LENGTH_FEET,
+    LENGTH_KILOMETERS,
+    LENGTH_METERS,
+    LENGTH_MILES,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util.distance import convert
+import homeassistant.util.dt as dt_util
+
+from .const import (
+    ATTR_ADDRESS,
+    ATTR_AT_LOC_SINCE,
+    ATTR_DRIVING,
+    ATTR_LAST_SEEN,
+    ATTR_PLACE,
+    ATTR_SPEED,
+    ATTR_WIFI_ON,
+    COMM_MAX_RETRIES,
+    COMM_TIMEOUT,
+    DOMAIN,
+    LOGGER,
+    SPEED_DIGITS,
+    SPEED_FACTOR_MPH,
+)
+
+
+class AccountData(TypedDict, total=False):
+    """Account data."""
+
+    api: Life360
+    coordinator: DataUpdateCoordinator
+    unsub: Callable[[], None]
+    re_add_entry: bool
+
+
+class IntegData(TypedDict):
```

Comment:
I recommend using a dataclass instead. It's more developer friendly.

https://docs.python.org/3/library/dataclasses.html

Reply:
I used TypeDict because I saw it used elsewhere. I assumed it was an accepted way. I guess I can go off and spend more time learning about dataclasses if you want.

Reply:
Done.